app/decorators/service_provider_session_decorator.rb

```diff
@@ -39,10 +40,14 @@ def sp_name
   end
 
   def sp_return_url
-    sp.return_to_sp_url
+    if (request_url = session[:request_url])
```

Comment:
Ok, I was using the convention of parens in an if-statement to indicate assignment.

I'll pull it into a helper method.